Sources/SwiftCoAP/SCClient.swift

```diff
@@ -313,8 +313,8 @@ class SCClient: NSObject {
         urlRequest.url = URL(string: urlString)
         urlRequest.timeoutInterval = SCMessage.kMaxTransmitWait
         urlRequest.cachePolicy = .useProtocolCachePolicy
-        
-        NSURLConnection.sendAsynchronousRequest(urlRequest as URLRequest, queue: OperationQueue.main) { (response, data, error) -> Void in
+
+        URLSession.shared.dataTask(with: urlRequest as URLRequest) { (data, response, error) -> Void in
```

Comment:
Not the best thing to use a shared `URLSession` but OK for now. The primary reason is to allow authenticated requests. Again, we can change it later to a user-provided `URLSession`.